### PR TITLE
update yunet to v2

### PR DESCRIPTION
**Pull request by Wwupup**
This PR requires https://github.com/opencv/opencv/pull/23020 and should be merged along with the next OpenCV release (probably 4.8.0).

### CODE REVIEW

**Comment by WanliZhong** [comment]
Thanks for your updating! There are something should be modifide
- Should we update the accuracy evaluation results in https://github.com/opencv/opencv_zoo/tree/master/models/face_detection_yunet? image
- If we change the model name, we should also modify the demo in sface. https://github.com/opencv/opencv_zoo/blob/master/models/face_recognition_sface/demo.py#L60)

File: models/face_detection_yunet/demo.cpp [hunk 1]
```diff
@@ -112,7 +112,7 @@ int main(int argc, char** argv)
     cv::CommandLineParser parser(argc, argv,
         "{help  h           |                                   | Print this message}"
         "{input i           |                                   | Set input to a certain image, omit if using camera}"
-        "{model m           | face_detection_yunet_2022mar.onnx | Set path to the model}"
+        "{model m           | face_detection_yunet_202303.onnx  | Set path to the model}"
```
Comment (fengyuentau) [inline models/face_detection_yunet/demo.cpp:115]: `202303` -> `2023mar`

File: models/face_detection_yunet/demo.py [hunk 2]
```diff
@@ -27,8 +27,8 @@
 parser = argparse.ArgumentParser(description='YuNet: A Fast and Accurate CNN-based Face Detector (https://github.com/ShiqiYu/libfacedetection).')
 parser.add_argument('--input', '-i', type=str,
                     help='Usage: Set input to a certain image, omit if using camera.')
-parser.add_argument('--model', '-m', type=str, default='face_detection_yunet_2022mar.onnx',
-                    help="Usage: Set model type, defaults to 'face_detection_yunet_2022mar.onnx'.")
+parser.add_argument('--model', '-m', type=str, default='face_detection_yunet_202303.onnx',
+                    help="Usage: Set model type, defaults to 'face_detection_yunet_202303.onnx'.")
```
Comment (fengyuentau) [inline models/face_detection_yunet/demo.py:31]: `202303` -> `2023mar`

File: models/face_detection_yunet/face_detection_yunet_202303.onnx
Comment (fengyuentau) [inline models/face_detection_yunet/face_detection_yunet_202303.onnx:1]: `202303` -> `2023mar`

**Review by fengyuentau — COMMENTED** [review]
Thank you 👍

Please use tools/quantize to quantize the model. Also the filename in tools/quantize/quantize-ort.py needs to be updated.

File: tools/eval/eval.py [hunk 3]
```diff
@@ -54,14 +54,14 @@
...
-        modelPath=os.path.join(root_dir, "models/face_detection_yunet/face_detection_yunet_2022mar.onnx"),
+        modelPath=os.path.join(root_dir, "models/face_detection_yunet/face_detection_yunet_2023mar.onnx"),
         topK=5000,
         confThreshold=0.3,
         nmsThreshold=0.45),
     yunet_q=dict(
         name="YuNet",
         topic="face_detection",
-        modelPath=os.path.join(root_dir, "models/face_detection_yunet/face_detection_yunet_2022mar-act_int8-wt_int8-quantized.onnx"),
+        modelPath=os.path.join(root_dir, "models/face_detection_yunet/face_detection_yunet_2023mar-act_int8-wt_int8-quantized.onnx"),
```
Comment (fengyuentau) [inline tools/eval/eval.py:64]: `face_detection_yunet_2023mar_int.onnx`

**Review by WanliZhong — APPROVED** [review]
👍👍👍

**Comment by fengyuentau** [comment]
Merge for now as 4.8.0 will be released soon. Benchmark will done in another PR.

**Pull request merged**